nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -191,17 +194,20 @@ def get_nodes_by_type(self, node_type: str) -> List[onnx.NodeProto]:
                 output.append(node)
         return output
 
-    def get_weight_tensor_name(self, node_name: str) -> str:
+    def get_weight_tensor_name(self, node: onnx.NodeProto) -> Optional[str]:
         # TODO(kshpv): add search of input weight tensor
         """
         Returns weight tensor name from the 1-index.
 
-        :param node_name: Name of the node.
+        :param node: node.
         :return: Weight tensor name.
         """
-        node_inputs = self.get_node_edge_names(node_name)['input']
-        weight_tensor = node_inputs[1]
-        return weight_tensor
+        metatype = ONNX_OPERATION_METATYPES.get_operator_metatype_by_op_name(node.op_type)
+        if metatype in WEIGHT_LAYER_METATYPES:
+            node_inputs = self.get_node_edge_names(node.name)['input']
+            weight_tensor = node_inputs[1]
+            return weight_tensor
+        return None
```

Comment:
In this case, we should check that `None` wouldn't be called further.

Reply:
Removed returning None